packages/components/src/components/rating-stars/rating-stars.tsx

```diff
@@ -9,298 +9,195 @@
+  @State() firstStarSelected: boolean;
+
+  /** Emitted when the rating has changed */
+  @Event() scaleChange: EventEmitter;
 
-  renderIcon = (color: string, size: string, selected?: boolean) => {
-    if (selected) {
-      return `<scale-icon-action-favorite color=${color} size=${size} selected />`;
-    } else {
-      return `<scale-icon-action-favorite color=${color} size=${size} />`;
-    }
-  };
   connectedCallback() {
-    statusNote({ source: this.hostElement, tag: 'beta' });
-    this.handleMouseEnter = this.handleMouseEnter.bind(this);
-    this.handleMouseLeave = this.handleMouseLeave.bind(this);
-    this.handleMouseMove = this.handleMouseMove.bind(this);
-    this.handleMouseClick = this.handleMouseClick.bind(this);
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-    this.handleTouchStart = this.handleTouchStart.bind(this);
-    this.handleTouchEnd = this.handleTouchEnd.bind(this);
-    this.handleTouchMove = this.handleTouchMove.bind(this);
+    this.firstStarSelected = false;
   }
 
-  componentDidLoad() {
-    handleListeners(this.element, 'addListeners');
+  // constructs the aria message for the current rating
+  getRatingText() {
+    const filledText = this.ariaLabelTranslation
+      .replace(/\$rating/g, `${this.rating}`)
+      .replace(/\$maxRating/g, `${this.maxRating}`);
+    return filledText;
   }
 
-  disconnectedCallback() {
-    handleListeners(this.element, 'removeListeners');
-  }
+  handleInput = (ev: InputEvent) => {
+    const input = ev.composedPath()[0] as HTMLInputElement;
+    const rating = input.value;
 
-  handleMouseLeave() {
-    this.isHovering = false;
-    return false;
-  }
-
-  handleMouseEnter() {
-    if (!this.disabled) {
-      this.isHovering = true;
-      return true;
-    }
-  }
-
-  handleMouseMove(event: MouseEvent) {
-    this.hoverValue = this.getValueFromXPosition(null, event);
-  }
-
-  handleMouseClick(event: MouseEvent) {
-    this.setValue(this.getValueFromXPosition(null, event));
-  }
-
-  handleKeyDown(event: KeyboardEvent) {
-    if (this.disabled) {
-      return;
+    if (this.rating === 0) {
+      this.rating = 1;
+    } else {
+      this.rating = Number(rating);
     }
 
-    if (event.key === 'ArrowRight') {
-      const valuePlus = this.value + this.precision;
-      this.value = clamp(valuePlus, 0, this.max);
-      this.emitRatingValue();
-      event.preventDefault();
-    }
+    this.scaleChange.emit({ value: this.rating });
+  };
 
-    if (event.key === 'ArrowLeft') {
-      const ratingMinus = this.value - this.precision;
-      this.value = clamp(ratingMinus, 0, this.max);
-      this.emitRatingValue();
-      event.preventDefault();
+  handleTouchEnd = (ev: TouchEvent) => {
+    const star = ev.composedPath()[0] as StarInterface;
+    const starValue = Number(star.dataset.value);
+    if (starValue === 1) {
+      if (this.firstStarSelected) {
+        this.rating = starValue;
+        this.firstStarSelected = !this.firstStarSelected;
```

Comment:
are you sure `firstStarSelected` needs to be `State` and not a simple private property?

Reply:
I have changed firstStarSelected to be a private property with a default value of false